10-dns-bootstrap.md

```diff
@@ -1,68 +1,136 @@
+    - MAY ignore the condition altogether (i.e. the seed filtering is best effort only).
+  - for `A` and `AAAA` queries:
+    - MUST return only nodes listening on the default port 9735, as defined in
+    [BOLT #1](01-messaging.md).
+  - for `SRV` queries:
+    - MAY return nodes that are listening on non-default ports, since `SRV`
+    records return a _(hostname,port)_-tuple.
+  - upon receiving a _wildcard_ query:
+    - MUST select a random subset of up to `n` IPv4 or IPv6 addresses of nodes
+    that are listening for incoming connections.
+  - upon receiving a _node_ query:
+    - MUST select the record matching the `node_id`, if any, AND return all
+    addresses associated with that node.
+
+Querying clients:
+  - MUST NOT rely on any given condition being met by the results.
 
 ### Reply Construction
 
-The results are serialized in a reply with a query type matching the client's query type, i.e. `A` queries result in `A` replies, `AAAA` queries result in `AAAA` replies, and `SRV` queries result in `SRV` replies, but they may be augmented with additional records (e.g. to add `A` or `AAAA` records matching the returned `SRV` records).
+The results are serialized in a reply with a query type matching the client's
+query type. For example, `A`, `AAAA`, and `SRV` queries respectively result in
+`A`, `AAAA`, and `SRV` replies. Additionally, replies may be augmented with
+additional records (e.g. to add `A` or `AAAA` records matching the returned
+`SRV` records).
+
+For `A` and `AAAA` queries, the reply contains the domain name and the IP
+address of the results.
 
-For `A` and `AAAA` queries, the reply contains the domain name and the IP address of the results.
-The domain name MUST match the domain in the query in order not to be filtered by intermediate resolvers.
+The domain name MUST match the domain in the query, in order not to be filtered
+by intermediate resolvers.
 
 For `SRV` queries, the reply consists of (_virtual hostnames_, port)-tuples.
-A virtual hostname is a subdomain of the seed root domain that uniquely identifies a node in the network.
+A virtual hostname is a subdomain of the seed root domain that uniquely
+identifies a node in the network.
 It is constructed by prepending the `node_id` condition to the seed root domain.
-The DNS seed MAY additionally return the corresponding `A` and `AAAA` records that indicate the IP address for the `SRV` entries in the Extra section of the reply.
-Due to the large size of the resulting reply, the reply may be dropped by intermediate resolvers, hence the DNS seed MAY omit these additional records upon detecting a repeated query.
 
-Should no entries match all the conditions then an empty reply MUST be returned.
+The DNS seed:
+  - MAY additionally return the corresponding `A` and `AAAA` records that
+  indicate the IP address for the `SRV` entries in the Extra section [FIXME: not sure what this refers to, is 'Extra section' defined elsewhere?] of the reply.
```

Comment:
Me too.... @cdecker ?

Reply:
The additional section in [RFC 1035](https://www.ietf.org/rfc/rfc1035.txt) allows us to return results whose type does not match the query type. This is commonly used to piggyback the `A` record together with the reply for an `MX` record. Otherwise the client would have to issue two queries (a) "give me the name of the server", and (b) "now tell me where to find that name". In our case `SRV` returns only names, not the matching addresses, so we just add them in the additional section.

Reply:
And yes, I misnamed the section in the original draft :-)